src/net_processing.cpp

```diff
@@ -1546,56 +1546,60 @@ PeerRef PeerManagerImpl::RemovePeer(NodeId id)
     return ret;
 }
 
-bool PeerManagerImpl::GetNodeStateStats(NodeId nodeid, CNodeStateStats& stats) const
+CNodeStateStats PeerManagerImpl::GetNodeStateStats(NodeId nodeid) const
 {
+    CNodeStateStats stats;
     {
         LOCK(cs_main);
         const CNodeState* state = State(nodeid);
-        if (state == nullptr)
-            return false;
-        stats.nSyncHeight = state->pindexBestKnownBlock ? state->pindexBestKnownBlock->nHeight : -1;
-        stats.nCommonHeight = state->pindexLastCommonBlock ? state->pindexLastCommonBlock->nHeight : -1;
+        if (state == nullptr) {
```

Comment:
Maybe I'm missing the context, but when would we expect to trigger this?
If this is called directly by user, why does it make sense to return defaults, rather than explicitly state the node is not found.

Reply:
Thanks for having a look.  This can be seen e.g. in the GUI Peers tab details area when the next three blocks fields display "Unknown" during the peer setup phase.

The idea is to return sensible defaults where it makes sense and potentially simplifies the code (both ours and that of API clients). For instance, the "addr_processed" and "addr_rate_limited" fields are member-initialized to 0 and may as well be returned during set up, as the peer has processed and limited no addresses yet. Likewise, returning an "inflight" field with an empty array seems just as logical and makes life simpler for us and for clients, than sometimes not returning the field in the RPC response. For "pingwait" we are only interested in non-zero values so a default of zero makes sense. For "minfeefilter" we need a default, and so on. 

Will update the documentation here as follows:
```diff
         if (state == nullptr) {
-            // Return member-initialized sensible defaults for the fields where
-            // it makes sense and simplifies the code.
+            // Peer is still being set up; return member-initialized sensible defaults
+            // for the node state fields where it makes sense and simplifies the code.
             return stats;
```